File: source/code/nodejs/authentication/create_directory_saml.js
var samlDirectory = {
  name : 'My SAML Directory',
  description: 'A Directory used for SAML Authorization',
  provider: {
    providerId: 'saml',
    ssoLoginUrl: 'https://yourIdp.com/saml2/sso/login',
    ssoLogoutUrl: 'https://yourIdp.com/saml2/sso/logout',
    encodedX509SigningCert: '-----BEGIN CERTIFICATE-----\n...Certificate goes here...\n-----END CERTIFICATE-----',
    requestSignatureAlgorithm: 'RSA-SHA256'
  }
};

client.createDirectory(samlDirectory, function (err, directory) {
  if (err) {
    return console.error(err);
  }

  console.log('SAML directory created!');
});

File: source/code/ruby/authentication/create_directory_saml.rb
directory = client.directories.create(
  name: random_directory_name,
  description: 'description_for_some_test_directory',
  provider: {
    provider_id: 'saml ',
    sso_login_url: 'https://yourIdp.com/saml2/sso/login',
    sso_logout_url: 'https://yourIdp.com/saml2/sso/logout',
    encoded_x509_signing_cert: '-----BEGIN CERTIFICATE-----\n...Certificate goes here...\n-----END CERTIFICATE----- ',
    request_signature_algorithm: 'RSA-SHA256'
  }
)
